eclipse-collections/src/main/java/org/eclipse/collections/impl/list/Interval.java

```diff
@@ -712,6 +712,7 @@ public boolean equals(Object otherList)
         return !listIterator.hasNext();
     }
 
+    @SuppressWarnings("PointlessNullCheck")
     private boolean intObjectEqual(int i, Object object)
     {
         return object == null || !(object instanceof Integer) || ((Integer) object).intValue() != i;
```

Comment:
The method is mis-named I guess. It's intObjectNotEqual really.

I believe the inspection is correct and should not be suppressed. null is not an instance of Integer, so the second clause makes the first clause redundant.

Reply:
yeah I figured that, but did not know if I should rename it to be more sensible and then fix the inspection. 
I can do that in this commit or have a separate commit, PR. Let me know. I have it fixed (without the suppression) in my local, will just have to push it.